cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -97,7 +97,10 @@ static std::vector<std::shared_ptr<DataType>> kDictionaryIndexTypes = kIntegerTy
 static std::vector<std::shared_ptr<DataType>> kBaseBinaryTypes = {
```

Comment:
It seems you should also change `StringToBoolean` and other similar functions to include view types as inputs?

Reply:
That's another set of PRs. I got sucked into a half-float rabbit hole trying to fix string-to-x casts before opening this PR.